### ISSUE TITLE
Searcher Bidding Complexity Due To PBS

### ISSUE THREAD

**Issue by quintuskilbourn** [flashbots/mev-boost#144]
In thinking about how the auction for bundle inclusion will change under PBS the following problem came to mind. 

There are effectively two auctions happening. One for bundle inclusion and one for block inclusion. Searchers know how to bid for bundle inclusion, however block inclusion is very different. One may be faced with a scenario in which the sum of the winning bids for bundle inclusion in a Flashbots block (flashblock?) is not sufficient to win the PBS auction, despite the block being the most valuable block produced. In order for the flashblock to make it on-chain, searchers must work together and bid more than they need for bundle inclusion to facilitate block inclusion. 

Below is an example of such a scenario (it may be a little unrealistic, but the problem is more general than the example). The example shows two distinct MEV opportunities being bid on for inclusion in the Flashblock. For simplicity, both opportunities are only pursued by two searchers, the superior of which has a good inkling to what the other can extract/bid.

![image](https://user-images.githubusercontent.com/47097067/172890653-5886e49f-b01f-4140-8f82-c7a031ffd556.png)
### This seems unrealistic
* No one will come close to creating blocks with a similar value to ours
    * This is a strong premise, we should have good reason to assume weak competition if we do
    - One can imagine something like Yuga requiring all NFT mints to go through a certain wallet.
- No way searchers have this much profit left after getting a bundle included
    - Long tail MEV
    - Exclusive order flow is a problem. For example, we recently spoke to Rook searchers who had no competition for block inclusion because of EOF (for which they didn’t have to pay that much).
    - PBS may end up being a game of inches
- This will be rare
    - Even if it is rare, there may be many such blocks in a row leading bad UX (e.g. if this arises from specific market conditions like a Coinbase wallet incentive)
    - With EOF it may not be.

### Why this is a problem

- Searchers face a lot of uncertainty. It is hard to know how much everyone else is willing to bid for block inclusion especially because of the following uncertainties:
    - What other bundles will be included in the flashblock
    - How highly other searchers value their bundles
    - How high other searchers will bid to include their blocks
    
    This provides incentive for searchers to **merge/form coalitions** to coordinate on total bid (centralisation)
    
- EOF of other block builder gains traction.

-------
#### Some discussion that already happened
On the point about searcher uncertainty: "and the argument is that this incurs a lot of overhead on searcher’s side, so the collective time they spent playing mind games because of the existence of 2 auctions instead of 1 is negative sum. So we should devise a better mechanism design?"

Response: "Yes negative sum bc of time wasted and revenue wasted on blocks that don’t make it because someone else is trying to game the system or on overbidding to compensate for potential slackers.
 May also see searchers invest more time in auctions that see them more reliably included/with more “fair” UX for searchers."

**Comment by jldunne** [comment]
Thanks for taking the time to write this up @quintuskilbourn! I have added some thoughts and questions below.

My understanding of this scenario is as follows:

1. Searchers bid to have their bundles included in a block. There is some upper limit on the value of their bids i.e. the overall value of the bundle.
2. The block builder will separately bid to have the block chosen by the proposer. The block builder has no upper limit on what they might bid to have their block included, as it is advantageous for them to have blocks included (winning blocks is good for business, maybe they profit from MEV, more blocks means more EOF). A soft limit might be the `sum(searcher’s bids) - profit`.
3. If the searchers are bidding to have their block included, it’s possible that the builder they submit to is asking for less bids than another builder, thus leading to a decreased likelihood of their block being included. This incentivises the searchers to monitor other bundles that are coming in (is this possible?), and collude with eachother to pay more (but not more than the overall value of their bundles), so that there is more space for the builder to win the proposer auction.
4. If a builder or group of searchers has access to EOF, then this leads to more opportunity for the block builder to increase their proposer bid, thus leading to more likely block inclusion, and probably more orderflow. This has a centralising effect at block building.

**TL;DR**
It’s still not 100% clear to me what the negative implications of this are. The most concerning one seems to be access to exclusive orderflow exacerbating occurrence of this scenario, leading to dominant builders emerging. However, I think this is still the least bad area for “centralisation” to occur, and it’s in line with the original design goals of PBS, that block building be outsourced to a competitive market.

More detailed comments:

> In thinking about how the auction for bundle inclusion will change under PBS the following problem came to mind.
> There are effectively two auctions happening. One for bundle inclusion and one for block inclusion. Searchers know how to bid for bundle inclusion, however block inclusion is very different. 

Yes, however I think it’s important to note that these are two independent auctions. The block builder’s bid is separate from the bids that the searchers are submitting. The builder will decide independently how much of the searcher bids to give to the proposer. In the absence of the protocol enforcing this relationship, It’s not clear to me that this will definitely converge at `proposer bid == sum of searcher bids - small profit`. In fact, I can think of a few scenarios where the block builder might like to pay significantly more than what they received to the proposer (e.g. Mateusz’s idea about allowing partial blocks, or some kind of promotion).

> One may be faced with a scenario in which the sum of the winning bids for bundle inclusion in a Flashbots block (flashblock?) is not sufficient to win the PBS auction, despite the block being the most valuable block produced.

I see what you are saying. However, in this case, couldn’t the builder conceivably pay more out of their own funds in order to get this block included, and then require higher bids from searchers for the next block? They might be more likely to do this as there are some advantages to them winning blocks apart from profiting from searcher bids.

> In order for the flashblock to make it on-chain, searchers must work together and bid more than they need for bundle inclusion to facilitate block inclusion.

This framing doesn't seem quite right to me. There is an upper bound of what the searchers bids will be (close to the value of the bundle) because presumably they won’t bother bidding unless it is profitable to them. Is the idea that collectively upping their bids gives the block builder more room to increase its own bid, and therefore include their block?

> No one will come close to creating blocks with a similar value to ours

Hmm, I think we are expecting other builders to emerge very quickly after the merge, and there is certainly no guarantee that FB will be the strongest 😁 

> This provides incentive for searchers to merge/form coalitions to coordinate on total bid (centralisation)

AFAIK it’s already very common for searchers to work in teams. Is the issue that disparate groups of searchers will collude at bidding time? Why is this a problem?

> EOF of other block builder gains traction.

This is maybe more of an issue. If a builder/searcher has access to EOF, they can submit bundles that will theoretically be more profitable, which might lead to this dynamic happening more, and if the builder acts rationally and increases the proposer bid to be equal to the upper limit of this sum(bids), then maybe this builder is more likely to win. Winning builders will be more likely to get more OF, and therefore become dominant?

**Comment by quintuskilbourn** [comment]
Thanks for a thorough response @jldunne. 

Yes I agree that if builders are willing to take a loss to have a block included then this situation is less of a problem. 

The point I was trying to make is less a criticism of PBS and more of a potential flaw in determining bundle inclusion by taking the highest bid (because valuable blocks may not make it on-chain), especially if for some reason the builder is unwilling to take a loss on any block (this could come up if someone is trying to build blocks in a fully decentralised way for example). 

With PBS and under the assumption that a builder isn't willing to take a loss on a block, having searchers bid for bundle inclusion in a block means that searchers need to estimate three quantities to maximise profit: (1) how much they should bid for inclusion in the block, (2) how much the block bid should be for on-chain inclusion and (3) how much other searchers will bid for their opportunities. 

If searchers formed a coalition they could reduce/eliminate having to deal with (3), but I guess you're right. Centralisation at the searcher level shouldn't lead to any censorship as long as transactions that aren't picked up by searchers make it into the block normally.

**Comment by barnabemonnot** [comment]
Just wanted to post some thoughts about this issue. The problem as I understand it is that the economically efficient block isn't realised due to the fragmentation of the auctions.

More specifically, you have two competitive auctions (`A` and `B`) going on at the same time. For each auction, we expect the second-price to be realised, respectively `A2` and `B2`. It is possible that the sum of the two second-prices `A2 + B2` is less than the value `C` (or bid) of a third party bidding a whole block, while the sum of the two "first-prices" `A1 + B1` (highest value for each bundle) is greater than the bid of the third party. In this case, economic efficiency is not realised as the value realised in the block is smaller than it could have been (`A1 + B1 > C > A2 + B2`)

What prevented efficiency? Namely, it seems to be an information problem. If all bidders in the bundle auctions were aware of the third party bid, they could profitably enter into an agreement with each other to make a total bid up to `C`, plus increment (assume increment is zero, tie is broken by including the two bundles). For instance, any two bids `a` and `b` such that `a+b = C` and `a >= A2` and `b >= B2` would be fine, and the bids are incentive compatible as long as `A1 >= a` and `B1 >= b`. If we have `A1 + B1 > C`, then the feasible set is not trivially empty. Each point in the set is also an equilibrium, as either player unilaterally bidding less means they both realise a payoff of zero (they are not included in the block).

Now we could assume 1) that neither player knows the other's value, or 2) that only one of them knows the other's, or 3) that they each know the other's. I am just writing some thoughts but it should be discussed :)

In 1), I wonder if it is possible to design a mechanism where they would reveal their true value. I am thinking some sort of VCG mechanism (https://en.wikipedia.org/wiki/Vickrey%E2%80%93Clarke%E2%80%93Groves_mechanism), but is it implementable?

In 2), it feels like the player who knows the other's value should be able to make them bid the entirety of their value or up to the amount necessary to beat the third-party bid, whichever is lower.

In 3), there may be ways to arrive at mutually-agreed upon "fair" agreements, where for instance each player bids an equal fraction of the profit they expect to make (they know how much the other stands to make since they know their values).

Then of course should the builder interacting with the two searchers wish to "top up" the searcher bids to beat the EOF builder, it is basically a subsidy, which may be rational in a larger model where the builder gets external rewards from having one of their blocks included.

**Comment by andy-t-wang** [comment]
I figure searchers will broadcast their transactions to all honest block builders. If the builder acts dishonestly and steals the MEV opportunities, then searchers will stop submitting to that builder. Thus, the builder will see their block value decrease as they lose valuable order flow. This means that honest block builders will generate the most of the blocks as they build the most valuable blocks and searchers should be fine submitting txns to all honest block builders.

**Comment by ziljah** [comment]
e i

**Comment by Napcalx** [comment]
So quick one I'm a bit lost on what EOF means also if two competing searchers propose bids that are close and the higher gets picked by the Builder and bundled with another auction won according to the image dropped in the initial post, if I understand well the bid from EOF would get priority over the two bids won through Auction or are they all added together to form a block and the Bid from coinbase doesn't pass through an Auction.

**Comment by quintuskilbourn** [comment]
EOF = exclusive order flow. I.e. order flow that is only available to one builder

I'm not sure I follow the rest of the question, but 
* only one block can be selected so they are mutually exclusive
* block builders aggregate bundles from searchers. Some of these bundles are mutually exclusive (e.g. they pursue the same opportunity) and some bundles are not so both can be included.

**Comment by Napcalx** [comment]
Alright, my second question was which block was going to be added between the block from the flashbot or the one from the competitor block since they are both going to the Mev Boost
